### PR TITLE
Additional data added for PLP events

### PR DESCRIPTION
**Pull request by miakobchuk**
In order to push data into appEventData we need to get it from PLP
Additional parameters were added to search-results-view and search-product-click

### CODE REVIEW

File: src/api/search.ts [hunk 1]
```diff
@@ -239,7 +239,7 @@ const getProductSearch = async ({
     window.adobeDataLayer.push((dl: any) => {
       dl.push({
         event: 'search-results-view',
-        eventInfo: { ...dl.getState(), searchUnitId: SEARCH_UNIT_ID },
+        eventInfo: { ...dl.getState(), searchUnitId: SEARCH_UNIT_ID, searchResults: results?.data?.productSearch },
```
Comment (chicharr) [inline src/api/search.ts:242]: Hey @miakobchuk ,
the event you need to track for PLP view is the one above `category-results-view`.
And when I look in the event-info I already see the search results data on it. 
What are you adding here as `searchResults` ?
Screenshot 2024-08-26 at 11 21 15

File: src/components/ProductItem/ProductItem.tsx [hunk 2]
```diff
@@ -200,6 +200,7 @@ export const ProductItem: FunctionComponent<ProductProps> = ({
         eventInfo: {
           ...dl.getState(),
           sku: product?.sku,
+          productView,
```
Comment (chicharr) [inline src/components/ProductItem/ProductItem.tsx:203]: Not sure if it's a good idea to pass the whole productView to the datalayer, or would be better just to extract the few field/attributes that we need

Comment (miakobchuk) [reply src/components/ProductItem/ProductItem.tsx:203]: Yes. I can extract only required attributes.

File: src/api/search.ts (hunk 1 above)
Comment (miakobchuk) [reply src/api/search.ts:242]: I am using productView from results as it contains some additional attributes, including: price, rating, season, model...

Comment (miakobchuk) [reply src/api/search.ts]: searchRequest data was replaced with specific attributes data

File: src/components/ProductItem/ProductItem.tsx (hunk 2 above)
Comment (miakobchuk) [reply src/components/ProductItem/ProductItem.tsx]: productView data was replaced with specific attributes data

**Review by chicharr — APPROVED** [review]

**Pull request merged**